cmd/skopeo/tags.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/containers/image/v5/docker"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+
+	"io"
+)
+
+// tagListOutput is the output format of (skopeo tags), primarily so that we can format it with a simple json.MarshalIndent.
+type tagListOutput struct {
+	DockerRepository string   `json:",omitempty"`
+	Tags             []string `json:",omitempty"`
+}
+
+type tagsOptions struct {
+	global *globalOptions
+	image  *imageOptions
+}
+
+func tagsCmd(global *globalOptions) cli.Command {
+	sharedFlags, sharedOpts := sharedImageFlags()
+	imageFlags, imageOpts := imageFlags(global, sharedOpts, "", "")
+
+	opts := tagsOptions{
+		global: global,
+		image:  imageOpts,
+	}
+
+	return cli.Command{
+		Name:  "list-repository-tags",
+		Usage: "List tags in a Docker v2 API repository specified by the REPOSITORY-NAME",
+		Description: fmt.Sprintf(`
+	Return the list of tags in from the repository in the Docker v2 registry "REPOSITORY-NAME"
+
+	See skopeo(1) section "REPOSITORY NAMES" for the expected format
+	`),
+		ArgsUsage: "REPOSITORY-NAME",
+		Flags:     append(sharedFlags, imageFlags...),
+		Action:    commandAction(opts.run),
+	}
+}
+
+func (opts *tagsOptions) run(args []string, stdout io.Writer) (retErr error) {
+	ctx, cancel := opts.global.commandTimeoutContext()
+	defer cancel()
+
+	if len(args) != 1 {
+		return errors.New("Usage: skopeo list-repository-tags <registry host>:<port>/<repository name>")
+	}
+
+	// Add a leading // to make it valid docker reference, even though user doesn't specify the transport
+	repoRef := fmt.Sprintf("//%s", args[0])
+
+	repo, err := docker.ParseReference(repoRef)
+	if err != nil {
+		return fmt.Errorf("Cannot parse repository reference: %v", err)
+	}
+
+	outputData := tagListOutput{
+		DockerRepository: repo.DockerReference().Name(),
+		Tags:             []string{},
+	}
+
+	sys, err := opts.image.newSystemContext()
+	if err != nil {
+		return err
+	}
+
+	outputData.Tags, err = docker.GetRepositoryTags(ctx, sys, repo)
+	if err != nil {
+		return fmt.Errorf("Error listing repository tags: %v", err)
+	}
+
+	out, err := json.MarshalIndent(outputData, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	_, err = fmt.Fprintf(stdout, "%s\n", string(out))
+
+	if err != nil {
+		return fmt.Errorf("Error marshalling json output from tag listing: %v", err)
```

Comment:
I suppose this belongs to `MarshalIndent`, the `Fprintf` error should have a different description. (It wouldn’t at all be unprecedented to just silently ignore the error, so just returning `err` without bothering with extra description would still be fine — as long as the description is not _misleading_.)